### DISCUSSION TITLE
Where are the specs or tests?

### DISCUSSION THREAD

**Discussion by wuarmin** [table-library/react-table-library#13] [General]
Hey!
Are there any tests or specs? I cannot find them? Is storybook used for testing? 

Thank you and Best regards

**Comment by rwieruch** [comment]
Storybook + Loki is used for Screenshot Tests: https://github.com/table-library/react-table-library/tree/master/.loki/reference I didn't implement any other tests, because my hope would be to get away with screenshot tests. At the moment, the screenshots are only taken after the initial render. It would be better if one would be able to change state (e.g. sort the table) and then take another screenshot. This is not possible with Loki, so the screenshot testing would have to move to Cypress + Applitools for instance.

**Reply by wuarmin** [reply]
Thank you for the response. Ok, I get it.  I don't know Loki and I haven't looked at screenshot-tests (I use React Testing Library) yet, but it sound interesting.

**Reply by rwieruch** [reply]
So the goal is to get away with only screenshot tests which add very low maintenance cost and which are very good suited for UI libraries. Loki is neatly integrated into Storybook, but not widely used and lacks some features such as changing the state of the component before taking a screenshot. As said, this would be better done with Cypress instead.
